Content.Server/StationEvents/Events/PowerGridCheck.cs

```diff
@@ -34,26 +44,50 @@ public override void Announce()
 
         public override void Startup()
         {
-            var entityManager = IoCManager.Resolve<IEntityManager>();
-
-            foreach (var component in entityManager.EntityQuery<ApcPowerReceiverComponent>(true))
+            foreach (var component in _entityManager.EntityQuery<ApcPowerReceiverComponent>(true))
             {
-                component.PowerDisabled = true;
                 _powered.Add(component.Owner);
             }
 
+            _numberPerSecond = Math.Max(1, (int)(_powered.Count / SecondsUntilOff)); // Number of APCs to turn off every second. At least one.
+
             base.Startup();
         }
 
-        public override void Shutdown()
+        public override void Update(float frameTime)
         {
-            var entMan = IoCManager.Resolve<IEntityManager>();
+            base.Update(frameTime);
+            _frameTimeAccumulator += frameTime;
+
+            var updates = 0;
+            if (_frameTimeAccumulator > UpdateRate)
+            {
+                updates = (int) (_frameTimeAccumulator / UpdateRate);
+                _frameTimeAccumulator -= UpdateRate * updates;
+            }
 
-            foreach (var entity in _powered)
+            for (var i = 0; i < updates; i++)
+            {
+                if (_powered.Count == 0)
+                    break;
+
+                var selected = _random.PickAndTake(_powered);
```

Comment:
Minor performance thing, but I guess it might matter for something like a list of all power receivers, where the list might be quite big: This should probably just shuffle the list initially, and then remove elements from the end of the list, rather than randomly removing from the middle.